addons/disposable/XEH_preInit.sqf

```diff
@@ -25,19 +25,21 @@ if (configProperties [configFile >> "CBA_DisposableLaunchers"] isEqualTo []) exi
     _unit call FUNC(changeDisposableLauncherClass);
 }] call CBA_fnc_addClassEventHandler;
 
-GVAR(NormalLaunchers) = [] call CBA_fnc_createNamespace;
-GVAR(LoadedLaunchers) = [] call CBA_fnc_createNamespace;
-GVAR(UsedLaunchers) = [] call CBA_fnc_createNamespace;
+GVAR(NormalLaunchers) = createHashMap;
+GVAR(LoadedLaunchers) = createHashMap;
+GVAR(UsedLaunchers) = createHashMap;
 GVAR(magazines) = [];
 GVAR(BackpackLaunchers) = createHashMap;
-GVAR(MagazineLaunchers) = [] call CBA_fnc_createNamespace;
+GVAR(MagazineLaunchers) = createHashMap;
 
 private _cfgWeapons = configFile >> "CfgWeapons";
 private _cfgMagazines = configFile >> "CfgMagazines";
 
 {
-    private _launcher = configName _x;
-    private _magazine = configName (_cfgMagazines >> (getArray (_cfgWeapons >> _launcher >> "magazines") select 0));
+    // Get case-sensitive config names
+    private _configLauncher = _cfgWeapons >> configName _x;
+    private _launcher = configName _configLauncher;
+    private _magazine = configName (_cfgMagazines >> (getArray (_configLauncher >> "magazines") select 0));
```

Comment:
```suggestion
    private _magazine = (compatibleMagazines (configName _x)) select 0;
```
might be possible to have a `magazine[]= {}` and just use mag wells, not likely
but it automatically gets config-case